server/src/test/java/io/crate/integrationtests/InsertIntoIntegrationTest.java

```diff
@@ -2018,4 +2018,33 @@ sub int as round((random() + 1) * 100)
             assertThat(response.rows()[0][0]).isEqualTo(generated);
         }
     }
+
+    @Test
+    public void test_insert_on_conflict_with_non_deterministic_column() {
+        execute("""
+            CREATE TABLE tbl (
+               id int PRIMARY KEY,
+               value DOUBLE PRECISION,
+               value_text TEXT,
+               rnd_col int GENERATED ALWAYS AS random() + 10
```

Comment:
Used `random` as it's not important which non-deterministic function we use to reproduce.

timestamp has issues when jdbc is enabled (https://github.com/crate/crate/pull/15303) - would have to select _raw to compare actual values.